test/functional/specs/Personalization/C28758.js

```diff
@@ -0,0 +1,144 @@
+import { t, ClientFunction } from "testcafe";
+import createNetworkLogger from "../../helpers/networkLogger";
+import { responseStatus } from "../../helpers/assertions/index";
+import createFixture from "../../helpers/createFixture";
+import addHtmlToBody from "../../helpers/dom/addHtmlToBody";
+import {
+  compose,
+  orgMainConfigMain,
+  debugEnabled
+} from "../../helpers/constants/configParts";
+import getResponseBody from "../../helpers/networkLogger/getResponseBody";
+import createResponse from "../../helpers/createResponse";
+import { TEST_PAGE as TEST_PAGE_URL } from "../../helpers/constants/url";
+import createAlloyProxy from "../../helpers/createAlloyProxy";
+import { injectInlineScript } from "../../helpers/createFixture/clientScripts";
+
+const networkLogger = createNetworkLogger();
+const config = compose(orgMainConfigMain, debugEnabled);
+const PAGE_WIDE_SCOPE = "__view__";
+
+createFixture({
+  title: "C28758 A VEC offer with ShadowDOM selectors should render",
+  url: `${TEST_PAGE_URL}?test=C28758`,
+  requestHooks: [networkLogger.edgeEndpointLogs]
+});
+
+test.meta({
+  ID: "C28758",
+  SEVERITY: "P0",
+  TEST_RUN: "Regression"
+});
+
+const shadowDomScript = `
+    const buyNowContent = \`
+    <div>
+      <input type="radio" id="buy" name="buy_btn" value="Buy NOW">
+      <label for="buy">Buy Now</label><br>
```

Comment:
Could you please extract these to a `/Personalization/constants` or in `functional/helpers/constants/`, to keep only the test code here.